bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/ProjectPerTargetProvisioningStrategy.java

```diff
@@ -111,28 +116,55 @@ public Map<BazelProject, CompileAndRuntimeClasspath> computeClasspaths(Collectio
             // populate map from result
             Map<BazelProject, CompileAndRuntimeClasspath> classpathsByProject = new HashMap<>();
             var aspectsInfo = new JavaAspectsInfo(result, workspace, aspects);
-            for (BazelProject bazelProject : bazelProjects) {
-                monitor.subTask(bazelProject.getName());
-                monitor.checkCanceled();
-
-                // build index of classpath info
-                var classpathInfo =
-                        new JavaAspectsClasspathInfo(aspectsInfo, workspace, availableDependencies, bazelProject);
-
-                // remove old marker
-                deleteClasspathContainerProblems(bazelProject);
 
-                // add the target
-                var problem = classpathInfo.addTarget(bazelProject.getBazelTarget());
-                if (!problem.isOK()) {
-                    createClasspathContainerProblem(bazelProject, problem);
+            // Performance optimization: Pre-load all dependency packages to avoid repeated Bazel queries
+            // This utilizes the existing batch optimization in BazelWorkspace.open()
+            try {
+                var packagesToPreload = collectDependencyPackages(bazelProjects, aspectsInfo, workspace);
+                if (!packagesToPreload.isEmpty()) {
+                    LOG.info(
+                        "Pre-loading {} dependency packages to optimize classpath computation",
+                        packagesToPreload.size());
+                    workspace.open(packagesToPreload);
                 }
+            } catch (Exception e) {
+                // Log warning but continue - classpath computation will still work, just slower
```

Comment:
What would be the exception here?
Why is it ok to continue? 

Assuming an exception is a broken/wrong Bazel binary, there is no point in continuing.
Assuming it's a NPE, this indicates a programming error. We should not continue but fix it.